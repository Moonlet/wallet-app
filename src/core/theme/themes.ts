export enum Theme {
    dark = 'dark',
    light = 'light'
}
