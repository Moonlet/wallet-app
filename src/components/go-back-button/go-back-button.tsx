import React from 'react';
import { TouchableOpacity } from 'react-native';
import { Text } from '../../library';

import stylesProvider from './styles';
import { withTheme } from '../../core/theme/with-theme';
import { Icon } from '../icon';

export interface IProps {
    onPress?: any;
    styles: ReturnType<typeof stylesProvider>;
}

export const GoBackButtonComponent = (props: IProps) => (
    <TouchableOpacity onPress={props.onPress} style={[props.styles.button]}>
        <Icon name="arrow-left-1" size={20} style={props.styles.icon} />
        <Text>Back</Text>
    </TouchableOpacity>
);

export const GoBackButton = withTheme(GoBackButtonComponent, stylesProvider);
