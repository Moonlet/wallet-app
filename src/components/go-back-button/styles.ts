import { StyleSheet } from 'react-native';
import { ITheme } from '../../core/theme/itheme';

export default (theme: ITheme) =>
    StyleSheet.create({
        button: {
            paddingHorizontal: 6,
            flexDirection: 'row'
        },

        icon: {
            color: theme.colors.accent,
            marginHorizontal: 6
        }
    });
