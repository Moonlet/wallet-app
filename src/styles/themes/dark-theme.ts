import { ITheme } from '../../core/theme/itheme';
import { COLORS } from '../colors';

export const darkTheme: ITheme = {
    dark: true,
    shadowGradient: ['#00000000', '#000000CC'],

    fontSize: {
        small: 12,
        regular: 16,
        large: 34
    },

    colors: {
        primary: COLORS.GRAY,
        accent: COLORS.AQUA,
        text: COLORS.WHITE,
        textDarker: COLORS.LIGHT_GRAY,
        positive: COLORS.HOT_GREEN,
        negative: COLORS.HOT_RED,

        cardBackground: COLORS.DARK_GRAY,
        appBackground: COLORS.COD_GRAY,
        settingsDivider: COLORS.BLACK
    }
};
