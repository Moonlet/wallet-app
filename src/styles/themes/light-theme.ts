import { ITheme } from '../../core/theme/itheme';
import { darkTheme } from './dark-theme';

export const lightTheme: ITheme = darkTheme;
