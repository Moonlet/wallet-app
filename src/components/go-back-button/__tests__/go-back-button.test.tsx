import React from 'react';
import { GoBackButtonComponent, IProps } from '../go-back-button';
import { darkTheme } from '../../../styles/themes/dark-theme';
import styleProvider from '../styles';

import { shallow } from 'enzyme';

const props: IProps = {
    styles: styleProvider(darkTheme)
};

describe('header go back button', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<GoBackButtonComponent {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
