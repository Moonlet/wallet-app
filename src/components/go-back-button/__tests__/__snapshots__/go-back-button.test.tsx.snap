// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header go back button renders correctly 1`] = `
"<TouchableOpacity onPress={[undefined]} style={{...}} activeOpacity={0.2}>
  <Icon name=\\"arrow-left-1\\" size={20} style={{...}} allowFontScaling={false} />
  <Component>
    Back
  </Component>
</TouchableOpacity>"
`;
